var/spack/repos/builtin/packages/costa/package.py

```diff
@@ -13,15 +13,20 @@ class Costa(CMakePackage):
     Based on the paper: https://arxiv.org/abs/2106.06601
     """
 
-    maintainers = ["haampie", "kabicm"]
+    maintainers = [
+        "haampie",
+        "kabicm",
+        "RMeli",
+    ]
     homepage = "https://github.com/eth-cscs/COSTA"
-    url = "https://github.com/eth-cscs/COSTA/releases/download/v2.0/COSTA-v2.0.tar.gz"
+    url = "https://github.com/eth-cscs/COSTA/archive/refs/tags/v2.0.tar.gz"
     git = "https://github.com/eth-cscs/COSTA.git"
 
     # note: The default archives produced with github do not have the archives
     #       of the submodules.
     version("master", branch="master", submodules=True)
-    version("2.0", sha256="de250197f31f7d23226c6956a687c3ff46fb0ff6c621a932428236c3f7925fe4")
+    version("2.1", sha256="c1e86452415083f7470b292d93ec60708b7c8dbafc2bac383636bb4b28135866")
+    version("2.0", sha256="0692a9c962282899894ccdbc7006216c835473a6999f1080c10bf94f23094521")
```

Comment:
Looks like `url_for_version` (line 38) is ensuring the sha256 is unchanged:

```
$ spack checksum costa 2.0 2.1
==> Found 2 versions of costa:
  
  2.1  https://github.com/eth-cscs/COSTA/archive/refs/tags/v2.1.tar.gz
  2.0  https://github.com/eth-cscs/COSTA/releases/download/v2.0/COSTA-v2.0.tar.gz

==> Fetching https://github.com/eth-cscs/COSTA/archive/refs/tags/v2.1.tar.gz
==> Fetching https://github.com/eth-cscs/COSTA/releases/download/v2.0/COSTA-v2.0.tar.gz

    version("2.1", sha256="c1e86452415083f7470b292d93ec60708b7c8dbafc2bac383636bb4b28135866")
    version("2.0", sha256="de250197f31f7d23226c6956a687c3ff46fb0ff6c621a932428236c3f7925fe4")
```

Probably better to remove `url` property and update `url_for_version`.

Reply:
`url_for_version` seems much better suited in this case, thanks for pointing it out.